src/cmd/builtin/config.c

```diff
@@ -11,9 +11,148 @@
+    for (int i = 0; i < sizeof (typemap) / sizeof (typemap[0]); i++) {
+        if (!strcasecmp (s, typemap[i].s)) {
+            *type = typemap[i].type;
+            *fsd_subtype = typemap[i].fsd_subtype;
+            return 0;
+        }
+    }
+    return -1;
+}
+
+static void print_object (json_t *o)
+{
+    if (json_is_string (o))
+        printf ("%s\n", json_string_value (o));
+    else {
+        char *s;
+        if (!(s = json_dumps (o, JSON_COMPACT | JSON_ENCODE_ANY)))
+            log_msg_exit ("error encoding json object");
+        printf ("%s\n", s);
+        free (s);
+    }
+}
+
+static void print_config_item (json_t *o,
+                               const char *path,
+                               json_type want_type,
+                               fsd_subtype_t fsd_subtype,
+                               optparse_t *p)
+{
+    json_type type;
+    double t;
+
+    if (path) {
+        if (!(o = jpath_get (o, path))) {
+            if (errno == ENOENT) {
+                if (optparse_hasopt (p, "default")) {
+                    printf ("%s\n", optparse_get_str (p, "default", NULL));
+                    return;
+                }
+                if (optparse_hasopt (p, "quiet"))
+                    exit (1);
+                log_msg_exit ("%s is not set", path);
+            }
+            log_err_exit ("%s", path);
+        }
+    }
+    type = json_typeof (o);
+    if (want_type == JSON_NULL // any
+        || (want_type == JSON_TRUE && type == JSON_FALSE) // boolean
+        || (want_type == type && fsd_subtype == FSD_NONE)) {
+        print_object (o);
+    }
+    else if (want_type == type
+        && fsd_subtype != FSD_NONE
+        && fsd_parse_duration (json_string_value (o), &t) == 0) {
+        if (fsd_subtype == FSD_INTEGER)
+            printf ("%d\n", (int)t);
+        else if (fsd_subtype == FSD_REAL)
+            printf ("%f\n", t);
+        else
+            printf ("%s\n", json_string_value (o));
+    }
+    else
+        log_msg_exit ("%s does not have the requested type",
+                      path ? path : "value");
+}
+
+static int config_get (optparse_t *p, int ac, char *av[])
+{
+    int optindex = optparse_option_index (p);
+    flux_t *h;
+    flux_future_t *f = NULL;
+    json_t *o;
+    const char *path = NULL;
+    const char *typestr;
+    json_type type;
+    fsd_subtype_t fsd_subtype;
+
+    if (optindex < ac)
+        path = av[optindex++];
+    if (ac - optindex > 0) {
+        optparse_print_usage (p);
+        exit (1);
+    }
+    if (!(h = builtin_get_flux_handle (p)))
+        log_err_exit ("flux_open");
+    if (!(f = flux_rpc (h, "config.get", NULL, FLUX_NODEID_ANY, 0))
+        || flux_rpc_get_unpack (f, "o", &o) < 0)
+        log_msg_exit ("Error fetching config object: %s",
+                      future_strerror (f, errno));
+    typestr = optparse_get_str (p, "type", "any");
+    if (parse_json_type (typestr, &type, &fsd_subtype) < 0)
+        log_msg_exit ("Unknown type");
```

Comment:
Minor suggestion: `log_msg_exit ("Unknown type: %s", typestr)`

Also, could add a test that ensures a bad type provided here fails with the expected error.

Reply:
> It would also be nice to have a command line interface to the "builtin" configuration values (i.e. `flux_conf_builtin_get(3)` - I wonder if this could be a `--builtin` flag on `flux config get` or if it should be a separate command.

Sure, good idea.  I'll add that too.